### ISSUE TITLE
[ISSUE] OPTIMIZE: Using external elasticsearch doesn't work if using https

### ISSUE THREAD

**Issue by lukas-beumer** [camunda/camunda-platform-helm#953]
**Describe the issue:**

With Camunda 8.3, the Optimize container has been changed and now defaults to a "migration" container.

This container uses the variables OPTIMIZE_ELASTICSEARCH_HOST and OPTIMIZE_ELASTICSEARCH_HTTP_PORT to connect to an Elasticsearch cluster.

Now, if we want to use an Elasticsearch cluster that is not hosted in the same Kubernetes cluster but in the cloud, for example, we get the following error because the Elasticsearch in the cloud only supports HTTPS:

│ Caused by: org.elasticsearch.client.ResponseException: method [GET], host [http://test123.west1.gcp.cloud.es.io:9243], URI [/], status line [HTTP/1.0 400 Bad Request]                                                                                                       
Client sent an HTTP request to an HTTPS server.

**Actual behavior:**

It's not possible to connect to an external hosted elasticsearch cluster with a https only url.

**Expected behavior:**

It should be possible to connect to an external hosted elasticsearch cluster with a https only url.

**How to reproduce:**

Use a external Elasticsearch Cluster and try to connect to it.

**Logs:**

> Caused by: org.elasticsearch.client.ResponseException: method [GET], host [http://test123.west1.gcp.cloud.es.io:9243], URI [/], status line [HTTP/1.0 400 Bad Request]                                                                                                       
Client sent an HTTP request to an HTTPS server.

**Environment: GCP (Elastic Cloud)

**Please note: Without the following info, it's hard to resolve the issue and probably it will be closed.**

- Platform: GCP
- Helm CLI version: 3.12.2
- Chart version: 8.3.0
- Values file: default one

**Comment by lukas-beumer** [comment]
@aabouzaid Is there any workaround we can implement yet?

**Comment by aabouzaid** [comment]
@lukas-beumer unfortunately, there is no workaround apart from Helm [post-rendering](https://helm.sh/docs/topics/advanced/#post-rendering), which I don't recommend now since we will have the 8.3.1 patch next week.

**Comment by lukas-beumer** [comment]
Okay thank you. I will wait for the fix then.

**Comment by aabouzaid** [comment]
@lukas-beumer could you please check if using these values works? (set both URL and proto/host/port).

```
global:
  elasticsearch:
    url: "https://es.external.com:9200" 
    protocol: https
    host: "es.external.com"
    port: 9200
```

**Comment by aabouzaid** [comment]
Fixed by ccf81f38027621c20c1c755ef58ca517951388b6 and e4b03e5095bb05cfef6da5e8262b289f9adc1de9
I will close this issue since it will be part of the next release.

**Comment by ross211** [comment]
I've been configuring ES authentication and SSL for Optimize using `environment-config.yaml` so I needed to modify `deployment.yaml` to pass `.Values.optimize.extraVolumeMounts` to migration init container too.

Can env vars be used to set the ES username, password, SSL and certificate_authorities options?

**Comment by lukas-beumer** [comment]
> I've been configuring ES authentication and SSL for Optimize using `environment-config.yaml` so I needed to modify `deployment.yaml` to pass `.Values.optimize.extraVolumeMounts` to migration init container too.
> 
> Can env vars be used to set the ES username, password, SSL and certificate_authorities options?

Same for us, would be a nice enhancement.

**Comment by aabouzaid** [comment]
Yes, with that change I've mentioned it will be possible for both the main container and the migration container.

**Comment by ross211** [comment]
That change adds support for env variables. I could not find any information on controlling the SSL options via env variables for Optimize which is why I was using environment-config.yaml.

Secondly `.Values.optimize.extraVolumeMounts` is not passed to the migration container like the main container so I'm not sure how I can load the SSL CA cert to the migration container unless there's an env var for that too?

**Comment by lukas-beumer** [comment]
@aabouzaid Do you have a release date for the 8.3.1?

**Comment by lukas-beumer** [comment]
@aabouzaid As @ross211 said, it is still not possible to call Elasticsearch environments inside initContainer via SSL/HTTPS because there is no way to add the corresponding extraVolumeMounts. I think that the issue here is not fully resolved yet.

**Comment by ross211** [comment]
@aabouzaid, I see there's a PR to address this: https://github.com/camunda/camunda-platform-helm/pull/1037

**Comment by hown3d** [comment]
#1036 adresses this issue in sort of the way.

**Comment by aabouzaid** [comment]
Thanks all for your input :raised_hands:
It's fixed by #1037

Also, worth mentioning that, since 8.3.1 it's possible to completely disable the migration init container and define yours in case the default migration container doesn't meet your needs.
